Project/Source/Modules/ModulePhysics.cpp

```diff
@@ -0,0 +1,396 @@
+							float3 newPos = contactPos - float3(contactManifold->getContactPoint(0).m_normalWorldOnB) * sphereCol->radius;
+							newPos.y = sphereCol->GetOwner().GetComponent<ComponentTransform>()->GetGlobalPosition().y;
+							sphereCol->GetOwner().GetComponent<ComponentTransform>()->SetGlobalPosition(newPos);
+						}
+						break;
+					}
+					case ComponentType::BOX_COLLIDER: {
+						ComponentBoxCollider* boxCol = (ComponentBoxCollider*) pbodyB;
+						boxCol->OnCollision();
+						if (moveBodyB) {
+							// TODO: Not implementing it since it is much more complicated and we won't use boxes as kinematic objects.
+						}
+						break;
+					}
+					case ComponentType::CAPSULE_COLLIDER: {
+						ComponentCapsuleCollider* capsuleCol = (ComponentCapsuleCollider*) pbodyB;
+						capsuleCol->OnCollision();
+						if (moveBodyB) {
+							float3 contactPos = pbodyA->GetOwner().GetComponent<ComponentTransform>()->GetGlobalPosition() + float3(contactManifold->getContactPoint(0).m_localPointA);
+							float3 newPos = contactPos - float3(contactManifold->getContactPoint(0).m_normalWorldOnB) * capsuleCol->radius;
+							newPos.y = capsuleCol->GetOwner().GetComponent<ComponentTransform>()->GetGlobalPosition().y;
+							capsuleCol->GetOwner().GetComponent<ComponentTransform>()->SetGlobalPosition(newPos);
+						}
+						break;
+					}
+					default:
+						break;
+					}
+				}
+			}
+		}
+	}
+	return UpdateStatus::CONTINUE;
+}
+
+UpdateStatus ModulePhysics::Update() {
+	if (App->time->IsGameRunning()) {
+		/*if (App->input->GetKey(SDL_SCANCODE_F1) == KS_DOWN) //TODO: DOnt do it by keyboard!!
+		debug = !debug;*/
+
+		if (debug == true) {
+			world->debugDrawWorld();
+		}
+
+		if (App->input->GetKey(SDL_SCANCODE_SPACE) == KS_DOWN) {
+			//btCollisionShape* colShape = new btSphereShape(2.f);
+			btCollisionShape* colShape = new btBoxShape(btVector3(1, 1, 1));
+
+			btVector3 localInertia(0, 0, 0);
+			colShape->calculateLocalInertia(3.0f, localInertia);
+
+			float3 position = App->camera->GetEngineCamera()->frustum.Pos();
+			btDefaultMotionState* myMotionState = new btDefaultMotionState(btTransform(btQuaternion::getIdentity(), btVector3(position.x, position.y, position.z)));
+			btRigidBody::btRigidBodyConstructionInfo rbInfo(3.0f, myMotionState, colShape, localInertia);
+			btRigidBody* body = new btRigidBody(rbInfo);
+
+			world->addRigidBody(body, WorldLayers::WORLD_ELEMENTS, WorldLayers::PLAYER | WorldLayers::WORLD_ELEMENTS | WorldLayers::EVERYTHING);
+			float3 f = App->camera->GetEngineCamera()->frustum.Front();
+			body->applyCentralImpulse(btVector3(f.x * 73.f, f.y * 73.f, f.z * 73.f));
+			/*
+
+
+
+			Sphere bullet;
+			bullet.color = Green;
+			bullet.radius = 1.0f;
+			bullet.SetPos(App->camera->Position.x, App->camera->Position.y, App->camera->Position.z);
+
+			App->physics->AddBody(bullet)->Push(-App->camera->Z.x * 20.0f, -App->camera->Z.y * 20.0f, -App->camera->Z.z * 20.0f);*/
```

Comment:
There is some commented code in this function.

Reply:
This code should not be included, it was just for testing. I'll remove it now.